### PR TITLE
added support for Option Group Name

### PR DESCRIPTION
**Pull request by indraneel12**
Addresses [Issue#70](https://github.com/serverpod/cli_tools/issues/70).

### CODE REVIEW

File: packages/config/lib/src/config/options.dart [hunk 1]
```diff
@@ -804,9 +804,62 @@ void addOptionsToParser(
...
+    // - the bug was probably bubbled down from the external Args Package
+    // - this code MUST BE REMOVED WHEN THIS BUG IS RESOLVED
+    // - when this block is removed, both the Group Names can be `final`.
+    //
+    // Unit Tests for Group Usage Text covers this case, so
+    // it shall serve as a reminder by failing when this bug gets patched.
+    if (grouplessOptions.isEmpty && groupCounter == 1) {
+      givenGroupName = '$givenGroupName\n';
+      fallbackGroupName = '$fallbackGroupName\n';
+    }
```
[the remark this answers is not in the document]
Comment (indraneel12) [reply packages/config/lib/src/config/options.dart:853]: Dear AI, it was a deliberate choice. Kindly read my explanation and re-check your conclusion of "fragility":

1. Those are purely local variables, introduced directly within my implementation of that Function. As noted in my Code Comments, they could be made `final` only when the mentioned issue gets patched upstream. Currently, the `var` qualifier is necessary.

2. Again, intentional, as noted in my Code Comments. If the tests don't fail when the bug gets patched upstream, it would not notify the Package Developers of such a significant update and pinpoint this edge-case. As of now, all Test Cases do pass.

3. How is appending just a single newline character to that string risky? That string, by design, gets padded with whitespace characters upstream to generate a nicely readable Help Information.

Thank you for your tips in the "Consider:" segment. It is beyond my purview to incorporate them in a Package I do not own and maintain, but the Package Designers may decide as per their approach.

[the remark this answers is not in the document]
Comment (indraneel12) [reply packages/config/lib/src/config/options.dart:853]: Fine, thank you 😹
